src/PointwiseFunctions/GeneralRelativity/WeylElectric.cpp

```diff
@@ -0,0 +1,89 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "PointwiseFunctions/GeneralRelativity/WeylElectric.hpp"
+
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "DataStructures/VectorImpl.hpp"
+#include "Utilities/ContainerHelpers.hpp"
+#include "Utilities/GenerateInstantiations.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/MakeWithValue.hpp"
+
+namespace gr {
+template <size_t SpatialDim, typename Frame, typename DataType>
+tnsr::ii<DataType, SpatialDim, Frame> weyl_electric(
+    const tnsr::ii<DataType, SpatialDim, Frame>& spatial_ricci,
+    const tnsr::ii<DataType, SpatialDim, Frame>& extrinsic_curvature,
+    const tnsr::II<DataType, SpatialDim, Frame>&
+        inverse_spatial_metric) noexcept {
+  tnsr::ii<DataType, SpatialDim, Frame> weyl_electric_part{
+      get_size(get<0, 0>(inverse_spatial_metric))};
+  weyl_electric<SpatialDim, Frame, DataType>(
+      make_not_null(&weyl_electric_part), spatial_ricci,
+      extrinsic_curvature, inverse_spatial_metric);
+  return weyl_electric_part;
+}
+
+template <size_t SpatialDim, typename Frame, typename DataType>
+void weyl_electric(
+    const gsl::not_null<tnsr::ii<DataType, SpatialDim, Frame>*>
+        weyl_electric_part,
+    const tnsr::ii<DataType, SpatialDim, Frame>& spatial_ricci,
+    const tnsr::ii<DataType, SpatialDim, Frame>& extrinsic_curvature,
+    const tnsr::II<DataType, SpatialDim, Frame>&
+        inverse_spatial_metric) noexcept {
+  if (UNLIKELY(get_size(get<0, 0>(*weyl_electric_part)) !=
+               get_size(get<0, 0>(inverse_spatial_metric)))) {
+    *weyl_electric_part = tnsr::ii<DataType, SpatialDim, Frame>(
+        get_size(get<0, 0>(inverse_spatial_metric)));
+  }
+  for (size_t i = 0; i < SpatialDim; ++i) {
```

Comment:
since Weyl = Ricci + other terms, you can avoid the size check (lines 36 to 40) and just start with
`*weyl_electric_part = spatial_ricci` before the for loop and omit line 43.